### ISSUE TITLE
Mark homework 1, exercise 3 of Anqi-Shi

### ISSUE THREAD

**Issue by anqis929** [anqis929/stat679#3]
# Hi Professor @cecileane here is the part 3 of my hw1. Thank you for giving us anther chance to do this.

https://github.com/anqis929/stat679/commit/a1f17365695ba828cd91d0c5cf393fb062e81ef2

**Comment by cecileane** [comment]
Thanks Irene! This is my way of asking Ruifeng (@ruifengchen) and Melody (@melpsain) to try your script and provide feedback on it.

**Comment by ruifengchen** [comment]
Dear Professor Ane,


May I ask that who else I hold responsibility to for providing feedback except Irene? Thank you.


Best,

Ruifeng

________________________________

**Comment by cecileane** [comment]
Definitely: Zachary Cohen. But he is having trouble with git. He submitted his work, but he will need to open a new issue. See our past discussion here:
https://github.com/kingcohn1/stat679work/issues/1
You could already clone his repository. His message on the first issue said that he is having trouble pushing his work for exercise 3 to github, and he copy-pasted his code on his issue message. I will keep you updated.
Cécile

On Oct 16, 2016, at 12:26 PM, ruifeng chen <notifications@github.com<mailto:notifications@github.com>> wrote:

Dear Professor Ane,


May I ask that who else I hold responsibility to for providing feedback except Irene? Thank you.


Best,

Ruifeng

________________________________

**Comment by ruifengchen** [comment]
OK! thank you.


Best,

Ruifeng

________________________________

**Comment by ruifengchen** [comment]
Hi Irene,
Excellent job! some feedback below.

You used a lot of command “grep”, “cut” “basename” and “sed”, these are the parts of significance we have learnt and what the course hopes us to do. Besides, you did very good by using the if function to find the last three values under3460, under3450 and under3440. It is very elegant to use “if” command in shell!

The script looks good, but unfortunately, when I tried to bash it, something goes wrong. I checked and found that in line 7, the “basenames” should be “basename” instead, please be careful next time to avoid these small but important typos.

In the part to find Nfail and the part to find frel, the code is not so bad, you wrote “sed -E 's/._=([0-9]+),._/\1/'” and “sed -E "s/._Rel=([0-9].[0-9]+._),ftol._/\1/"” respectively, but I believe there should be a space after “=” when you were trying to find the right value of Nfail, and there should be a space before ftol in the second part to find the right value of frel. You used “grep” first, that’s great, but please look carefully at the line you get after “grep” and then use the command “sed” . The similar thing happened when I looked into the command you wrote to find seed values, you wrote “sed -E "s/._seed=([0-9]+)/\1/"”, instead, the “=” after seed should be a space. It is the same problem, just please be careful at those small parts and make sure what the problem in fact wants.

If possible, please make your readme file more readable next time, add some links would be helpful for someone else to go directly to what you have done! Besides, more detailed comments would make your script more readable.

Good job again! Thank you!

**Comment by cecileane** [comment]
Thank you Ruifeng for your debugging work!

Regarding the base name of files, if I may chime it:
- it would be best for you to use the command `basename` instead of using parameter expansion. It's a great command to get used to. Very useful in many cases, so worth the extra effort to get some practice with it.
- I strongly discourage you to use a variable name that is the same as a command name. It might not cause a bug here, it could in other circumstances. So choose a name different from "basename" for the variable that holds the root name of your file. I'm speaking from experience here. I remember spending several hours to debug a problem that was caused by a variable I had named like an existing command.

Irene: please try to fix the bugs that Ruifeng found, so that Melody doesn't run into the same bugs. After you apply the suggestions from Ruifeng, commit/push your work then post a message here to let Melody know about your update. She will be able to get them with a `git pull` command.

**Comment by melpsain** [comment]
Hi Irene,

I just wanted to let you know that I am waiting on your bug fixes to provide you with some feedback!

Thanks,
Melody

**Comment by melpsain** [comment]
Hi Irene,

I  have some suggestions to help with your documentation and to help users navigate through your folders. Your first README.md file does not provide any information about what is contained in this stat679 directory. Therefore you could add some text about what your hw1 folder contains. Within your hw1 folder it would be nice if you created sub-directories for your scripts and figures (hw1EX2.csv). Creating these folders will help with organization of data. I would suggest that in your summarizeSNaQres.sh script it would be nice to let the user know where the code should be ran from to perform the task of collecting the information from the files. I like that you have used some comments in your script to guide the user. To make your code more "human readable" I would suggest incorporating comments that describe what is actually happening in the code and how certain commands such as sed are being used. 

To help with troubleshooting your code as you are writing them you could print the output to the screen to see if the code is producing the result you intended. 

I hope these comments help and thank you for letting me view your work!

Melody

**Comment by cecileane** [comment]
Irene,
In your new version of your summarizing script, you have 2 loops over data files, one nested inside the other:

``` shell
for filename in log/*.log # on line 4
do
  analysis=`grep rootname $filename | cut -f 2 -d ':'`
  basenames="${filename%.*}" #extract roots
  basename="${basename##*/}"  #get rid of "log/" in the front of roots
  outname="${basename}.out" #set roots name for *.out files
...
for filename in log/*.log # on line 37
do
...
done           # on line 89
...
done           # on line 112
```

That's a new bug.

I also see that you did not use `basename`. You should learn to use it, it's a very useful command.
- In your output file, you are also missing a comma between the "frel" and the "xabs"
- and you have spaces that should not be there: commas are used to separate columns, so there should be no spaces after commas

Melody: your last comments suggest that Irene's script didn't work for you. Is that correct?

**Comment by anqis929** [comment]
Hi Melody,
Thanks for your email, I’ve fixed the bug,

Thanks,
Irene
On Oct 23, 2016, at 17:48, Melody Sain <notifications@github.com<mailto:notifications@github.com>> wrote:

Hi Irene,

I just wanted to let you know that I am waiting on your bug fixes to provide you with some feedback!

Thanks,
Melody

**Comment by melpsain** [comment]
Hi Irene,

I was able to get your updated code to run correctly. (I removed the nested loop when I ran it on my machine). 

Thanks for fixing the bugs!

Melody
